node/src/components/gossiper/tests.rs

```diff
@@ -532,3 +533,223 @@ async fn should_timeout_gossip_response() {
 
     NetworkController::<NodeMessage>::remove_active();
 }
+
+#[tokio::test]
+async fn should_timeout_new_item_from_peer() {
```

Comment:
Nit-like suggestion for one more test :)

Please consider adding a similar one which ensures that `CheckItemReceivedTimeout` is **not** triggered before `VALIDATE_AND_STORE_TIMEOUT` time passes.

Reply:
As discussed offline, the pre-exisiting tests cover this case.